include/libpsl.h.in

```diff
@@ -40,12 +40,16 @@
 #define PSL_VERSION_PATCH @LIBPSL_VERSION_PATCH@
 #define PSL_VERSION_NUMBER @LIBPSL_VERSION_NUMBER@
 
+#ifndef __has_declspec_attribute
+#  define __has_declspec_attribute(x) 0
+#endif
+
 #ifndef PSL_API
 #if defined BUILDING_PSL && HAVE_VISIBILITY
 #  define PSL_API __attribute__ ((__visibility__("default")))
-#elif defined BUILDING_PSL && defined _MSC_VER && !defined PSL_STATIC
+#elif defined BUILDING_PSL && (defined _MSC_VER || (__has_declspec_attribute(dllexport) && __has_declspec_attribute(dllimport))) && !defined PSL_STATIC
```

Comment:
Why check for `dllimport` here if we don't actually use it in the branch? (similar below for `dllexport`)

Reply:
@donny-dont Thanks for the contribution. But @daurnimator seems to be right. Please fix up the checks and then it's ready to merge.